include/zephyr/rtio/rtio.h

```diff
@@ -1230,6 +1232,10 @@ __syscall void rtio_release_buffer(struct rtio *r, void *buff, uint32_t buff_len
 
 static inline void z_impl_rtio_release_buffer(struct rtio *r, void *buff, uint32_t buff_len)
 {
+	ARG_UNUSED(r);
+	ARG_UNUSED(buff);
+	ARG_UNUSED(buff_len);
+
```

Comment:
This needs to go in the `#else` block since these arguments  _are_ actually used if `CONFIG_RTIO_SYS_MEM_BLOCKS` is enabled.